tests/ops/op_math/test_pow_op.py

```diff
@@ -0,0 +1,612 @@
+            assert isinstance(op, qml.SX)
+            assert op.wires == qml.wires.Wires(0)
+
+    def test_decomposition_float_power(self):
+        """Test that the decomposition raises an error if no shortcut exists and the exponent is a float."""
+        base = qml.PauliX(0)
+        op = Pow(base, 0.11)
+
+        with pytest.raises(DecompositionUndefinedError):
+            op.decomposition()
+
+
+class TestInverse:
+    """Test the interaction between in-place inversion and the power operator."""
+
+    def test_base_already_inverted(self):
+        """Test that if the base is already inverted, then initialization un-inverts
+        it and applies a negative sign to the exponent."""
+        base = qml.S(0).inv()
+        op = Pow(base, 2)
+
+        assert base.inverse is False
+
+        assert op.z == -2
+        assert op.name == "S**-2"
+        assert op.base_name == "S**-2"
+        assert op.inverse is False
+
+    def test_invert_pow_op(self):
+        """Test that in-place inversion of a power operator only changes the sign
+        of the power and does not change the `inverse` property."""
+        base = qml.S(0)
+        op = Pow(base, 2)
+
+        op.inv()
+
+        assert base.inverse is False
+
+        assert op.z == -2
+        assert op.name == "S**-2"
+        assert op.base_name == "S**-2"
+        assert op.inverse is False
+
+    def test_inverse_setter(self):
+        """Assert that the inverse can be set to False, but trying to set it to True raises a
+        NotImplementedError."""
+        op = Pow(qml.S(0), 2.1)
+
+        op.inverse = False
+
+        with pytest.raises(NotImplementedError):
+            op.inverse = True
+
+
+class TestOperationProperties:
+    def test_basis(self):
+
+        base = qml.RX(1.2, wires=0)
+        op = Pow(base, 2.1)
+
+        assert base.basis == op.basis
+
+    def test_control_wires(self):
+
+        base = qml.Toffoli(wires=(0, 1, 2))
+        op = Pow(base, 3.5)
+
+        assert base.control_wires == op.control_wires
+
+
+class TestIntegration:
+    """Test the execution of power gates in a QNode."""
+
+    @pytest.mark.parametrize(
+        "diff_method", ("parameter-shift", "finite-diff", "adjoint", "backprop")
+    )
+    @pytest.mark.parametrize("z", (2, -2, 0.5))
+    def test_gradient_pow_rx(self, diff_method, z):
+        @qml.qnode(qml.device("default.qubit", wires=1), diff_method=diff_method)
+        def circuit(x, z):
+            Pow(qml.RX(x, wires=0), z)
+            return qml.expval(qml.PauliY(0))
+
+        x = qml.numpy.array(1.234, requires_grad=True)
+
+        expected = -np.sin(x * z)
+        assert qml.math.allclose(circuit(x, z), expected)
+
+        grad = qml.grad(circuit)(x, z)
+        expected_grad = -z * np.cos(x * z)
+        assert qml.math.allclose(grad, expected_grad)
+
+    @pytest.mark.xfail
+    def test_non_decomposable_power(self):
+        """This test will fail until we improve device support for power operators."""
```

Comment:
Just curious: what would be required to be done on the device level? How come the previous test case works?

Reply:
Manually specifying `"SX**-1"` in the list of supported operations. This is why I'd *love* to overhaul how we do device support expansion and validation.